dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,353 @@
     }
 
+    
 
-    /**
-     * @deprecated  As of 2016-10-12, replaced by {@link #snapshotIndex(HttpServletRequest,HttpServletResponse ,String)}
-     */
+
+    @Deprecated
     @PUT
     @Path("/create/{params:.*}")
     @Produces("text/plain")
     public Response createIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
         try {
-            InitDataObject init=auth(params,httpServletRequest, httpServletResponse);
+            InitDataObject init=auth(httpServletRequest, httpServletResponse);
 
             int shards=Integer.parseInt(init.getParamsMap().get("shards"));
             boolean live = init.getParamsMap().containsKey("live") ? Boolean.parseBoolean(init.getParamsMap().get("live")) : false;
             String indexName = init.getParamsMap().get("index");
 
-            indexName = create(indexName, shards, live);
+            if(indexName == null)
+                indexName=ContentletIndexAPIImpl.timestampFormatter.format(new Date());
+            indexName = (live) ? "live_" + indexName : "working_" + indexName;
+
+            APILocator.getContentletIndexAPI().createContentIndex(indexName, shards);
 
             return Response.ok(indexName).build();
-        } catch (DotSecurityException sec) {
-            SecurityLogger.logInfo(this.getClass(), "Access denied on createIndex from "+httpServletRequest.getRemoteAddr());
-            return Response.status(Status.UNAUTHORIZED).build();
         } catch (Exception de) {
             Logger.error(this, "Error on createIndex. URI: "+httpServletRequest.getRequestURI(),de);
             return Response.serverError().build();
         }
     }
-
+    
+    /**
+     * use modIndex instead, e.g. PUT /api/v1/esindex/{indexName}?action=clear
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param params
+     * @return
+     */
+    @Deprecated
     @PUT
     @Path("/clear/{params:.*}")
-    public Response clearIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
-        try {
-            InitDataObject init=auth(params,httpServletRequest,httpServletResponse);
-            String indexName = this.indexHelper.getIndexNameOrAlias(init.getParamsMap(),"index","alias",this.indexAPI);
-            if(UtilMethods.isSet(indexName)) {
-                APILocator.getESIndexAPI().clearIndex(indexName);
-            }
-            return Response.ok().build();
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to clear index: " + e.getMessage(), e);
-            return ResponseUtil.mapExceptionResponse(e);
-        }
-    }
+    public Response clearIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) throws DotDataException, IOException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'httpServletRequest' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)